src/shared/Core/Authentication/AuthenticationBase.cs

```diff
@@ -110,28 +110,44 @@ protected void ThrowIfTerminalPromptsDisabled()
             }
         }
 
-        protected bool TryFindHelperExecutablePath(string envar, string configName, string defaultValue, out string path)
+        protected bool TryFindHelperCommand(string envar, string configName, string defaultValue, out string command, out string args)
+        {
+            if (TryFindHelperPath(envar, configName, defaultValue, out string path))
+            {
+                if (path.EndsWith("dll"))
+                {
+                    command = $"dotnet";
+                    args = $" exec {path} ";
```

Comment:
For added safety, end the args with `-- ` to ensure that no args that callers may add can be confused as `dotnet` args.

`--` is used to signify the end of options; the rest are always passed straight to the child process.

https://learn.microsoft.com/en-us/dotnet/standard/commandline/syntax#the----token

```suggestion
                    args = $" exec {path} -- ";
```

Reply:
ignore me.. this isn't true for `dotnet exec` it seems!